Wdlupdate

bshifaw opened the pull request:
- Added description to ReadMe
- Made gvcf optional and default be vcf
- Added docker default variables
- Added automatic disksize estimation
- Updated haplotypecaller command to match production

Note: The workflow names have not changed to make it easier to compare in git. However, since the workflow is set to make VCF by default, before the merging I'll change the name of the files to 
haplotypecaller-**vcf**-gatk3.b37.wgs.inputs.json
haplotypecaller-**vcf**-gatk3.wdl

Also a note was added in the readme indicating a gvcf workflow will not be made available.

Commit:
Added description to ReadMe, made gvcf optional and default be vcf

Commit:
added docker default variables, made gvcf optional, added automatic disksize estimation

Commit:
minor correction

ldgauthier commented on haplotypecaller-gvcf-gatk3.b37.wgs.inputs.json:
```diff
@@ -14,24 +14,34 @@
   "##_COMMENT8": "MEMORY ALLOCATION", 
-  "HaplotypeCallerGvcf_GATK3.HaplotypeCaller.mem_size": "10 GB",
-  "HaplotypeCallerGvcf_GATK3.MergeVCFs.mem_size": "3 GB",
+  "#HaplotypeCallerGvcf_GATK3.MergeVCFs.machine_mem_gb": "Int? (optional)",
```
Logically I prefer this the original way since HaplotypeCaller gets called first.

bshifaw replied:
No, problem. I'll correct the order throughout the json.

ldgauthier commented on haplotypecaller-gvcf-gatk3.wdl:
```diff
@@ -48,112 +70,130 @@ workflow HaplotypeCallerGvcf_GATK3 {
-      -o ${gvcf_name} \
+      -o ${output_filename} \
       -L ${interval_list} \
       -ip ${default=100 interval_padding} \
```
Where did this come from? It's not in the production version: https://github.com/broadinstitute/dsde-pipelines/blob/master/genomes_in_the_cloud/single_sample/PairedSingleSampleWf.wdl This could throw errors.  If you have adjacent intervals in your input intervals then there could be repeated variants between shard x and shard x+1 that will lead to non-decreasing positions (and duplicates) when the shards are merged.  HaplotypeCaller has default 100bp padding on the assembly region and a default minimum of 50bp assembly region size anyway.

bshifaw replied:
Not sure. 
I'll switch over to the provided commands and to the gitc docker. 
Thanks

ldgauthier replied:
The GATK3 docker you have should be fine and your command just needs the -ip line to be dropped. It's better to use the max_alternate_alleles from production too because that will speed joint calling up for WGS in gross indel regions.

variant_index_parameter and variant_index_type are redundant provided your GVCF has a .g.vcf (or .g.vcf.gz) extension.  It's on my list to update production.

bshifaw replied:
I'll remove the redundant parameters and set the max_alt to 3. 
Since I don't know where the original command came from its probably a good idea to use the one from production. It also helps keep things consistent so i'll use the docker and commands from production.

Commit:
updated haplotypecaller command to production, switched to gitc docker

Commit:
- Set max_alt_all to 3
- Removed -variant_index_parameter 128000 and -variant_index_type LINEAR.
As they are redundant given GVCF has a .g.vcf (or .g.vcf.gz) extension

Commit:
rename from gvcf to vcf
